app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,131 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  before_action :ensure_course_open, :ensure_not_registered, :update_if_exist, only: :create
+
+  def index #:nodoc:
+    @staff_requests = @registrations.staff
+    @student_requests = @registrations.student
+  end
+
+  def create #:nodoc:
+    @registration.user = current_user
+    @registration.save!
+    @registration.update_attributes(registration_params)
+
+    # TODO: Notify lecturer
+    redirect_to course_path(@course), flash: { notice: t('.request_format') %
```

Comment:
Please do not use exceptions as part of flow control. Java has this problem, and it makes code hard to reason since there is no clue where the exception will be handled. Use save and valid?

Keep exceptions for exceptional situations.